.gitmodules

```diff
@@ -63,5 +67,6 @@
 	url = https://github.com/aspnet/common
 [submodule "src/sdk"]
 	path = src/sdk
-	url = https://github.com/dotnet/sdk.git
+	url = https://github.com/dotnet/sdk
 	branch = release/2.0.0
+	sourceBuildAutoUpdate = false
```

Comment:
Is this added just for illustration purposes?  Or is there a reason `sdk` shouldn't be auto updated, but everything else should be.

Reply:
There are currently a lot of `sdk` commits on that branch that aren't in source-build, and I didn't want to include them as part of this change (and the coming auto-updates). After this PR, another PR can be submitted to bring up `sdk` and enable auto-updates for it.

Reply:
But all the other repos are fine?

Reply:
The only other ones that have `branch`es are `coreclr`, `corefx`, `core-setup`, and `standard`, which are all up to date right now except `corefx` (which Wes has been working on).

Do you think it would be better to make this an explicit enable per-repo? I suppose it's possible that someone might add a `branch` without intending it to be auto-upgraded.

Reply:
Why not just have the repo's that are fixed and not currently updatable just point to a sha (assuming the branch property can point to a sha instead).

Reply:
That would lose info about what branch it is "sort of tracking" or "manually tracking", which seems useful to me to keep around so we don't start tracking the wrong branch at some point if we have to disable a branch and enable it again.

Implementation-wise, as-is, this is used with `git ls-remote refs/heads/{branch}`, and commits aren't heads on GitHub so VersionTools gets back a blank response. This takes extra time to look up, and in the end throws an exception. I would need to add a special case at some point in the chain to handle this being set to a commit. (Fortunately GitHub doesn't allow branches to be named in a way that looks like a commit.)

I don't know if anyone plans to use this, but that also wouldn't work with `git submodule update --remote`.

Reply:
Good points I'm fine with adding extra property assuming it doesn't mess up git up itself or git doesn't throw them out when this file gets updated for other means.

Reply:
Cool, yeah, existing config is preserved and doesn't interfere with anything.

I'm looking at bringing the target that creates submodule updaters into buildtools, and that makes me think the `sourceBuildAutoUpdate` config should be opt-in, not opt-out, to make the experience more intentional by default.